.github/workflows/build-macos.yml

```diff
@@ -203,41 +113,51 @@ jobs:
         with:
           submodules: true
 
-      - name: Download "pg_install/v14" artifact
-        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.0
+      - name: Get pg revisions for restoring cache
+        id: pg_rev
+        run: |
+          echo pg_rev_v14=$(git rev-parse HEAD:vendor/postgres-v14) | tee -a "${GITHUB_OUTPUT}"
+          echo pg_rev_v15=$(git rev-parse HEAD:vendor/postgres-v15) | tee -a "${GITHUB_OUTPUT}"
+          echo pg_rev_v16=$(git rev-parse HEAD:vendor/postgres-v16) | tee -a "${GITHUB_OUTPUT}"
+          echo pg_rev_v17=$(git rev-parse HEAD:vendor/postgres-v17) | tee -a "${GITHUB_OUTPUT}"
+
+      - name: Restore "pg_install/v14" cache
+        uses: actions/cache/restore@v4
         with:
-          name: pg_install--v14
           path: pg_install/v14
+          key: v1-${{ runner.os }}-${{ runner.arch }}-${{ env.BUILD_TYPE }}-pg-install-v14-${{ steps.pg_rev.outputs.pg_rev_v14 }}-${{ hashFiles('Makefile') }}
+          fail-on-cache-miss: true
 
-      - name: Download "pg_install/v15" artifact
-        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.0
+      - name: Restore "pg_install/v15" cache
+        uses: actions/cache/restore@v4
         with:
-          name: pg_install--v15
           path: pg_install/v15
+          key: v1-${{ runner.os }}-${{ runner.arch }}-${{ env.BUILD_TYPE }}-pg-install-v15-${{ steps.pg_rev.outputs.pg_rev_v15 }}-${{ hashFiles('Makefile') }}
+          fail-on-cache-miss: true
 
-      - name: Download "pg_install/v16" artifact
-        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.0
+      - name: Restore "pg_install/v16" cache
+        uses: actions/cache/restore@v4
```

Comment:
```suggestion
        uses: actions/cache/restore@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
```

Reply:
ok committed that.

There are a few of these in other workflows, should they be changed too?

```
$ git grep "cache@v4"
.github/actions/allure-report-generate/action.yml:      uses: actions/cache@v4
.github/actions/run-python-test-set/action.yml:      uses: actions/cache@v4
.github/workflows/periodic_pagebench.yml:      uses: actions/cache@v4
```